### PR TITLE
Interactive emote fixes

### LINKED ISSUE #15041: Interactive Emotes are Not Working
### Description

I tried to handshake someone, but they are not handshaking me.

### Test Merges

#14942: 'Warrior changes - minor reworks and rebalancing' by lKiseki at commit 1cfe22d05a 
#14975: 'SMG-45 HP->Squashhead, new sound' by Pariah919 at commit afc29b307d 
#14957: 'Fancy announcements (again)' by Neocloudy at commit 9ca9d5bb8e 

### Reproduction Steps

1. Aim either arm or hand in help intent for either handshake or fist bump respectively.
2. Click on person you want to handshake or fist bump.
3. They don't get prompt to handshake or fist bump.
4. Click on yourself.
5. You get your own prompt.
6. guh

### Screenshots

https://youtu.be/JlP65h_q4Ck

### PR DESCRIPTION
**Pull request by XElectricX**
## About The Pull Request

Fixes #15041:
- Was adding the interaction to the list on the wrong mob, oops
- Clashing with hugs and headpats, now use a TGUI selection system when right clicking on help intent

And while it may be a good candidate for radials, you people know I cannot sprite so just use the list.

## Why It's Good For The Game

We need our funny emotes working.

## Changelog
:cl:
fix: Fix interactive emotes
/:cl:

### CODE REVIEW

**Comment by XElectricX** [comment]
Literally the same branch I was using what is wrong with you git.

File: code/_onclick/click.dm [hunk 1]
```diff
@@ -355,6 +355,11 @@ if(selected_ability.target_flags & flagname && !istype(A, typepath)){\
 			return FALSE
 		if(COMSIG_MOB_CLICK_HANDLED)
 			return TRUE
+
+	//Any mob can begin an interaction when right clicking on help intent
+	if(ismob(A) && a_intent == INTENT_HELP && Adjacent(A) && interaction_emote(A))
```
Comment (ivanmixo) [inline code/_onclick/click.dm:360]: In the.. .fucking `RightClickOn`. Really?

**Review by ivanmixo — CHANGES_REQUESTED** [review]

**Review by ivanmixo — APPROVED** [review]
Prob fine

**Comment by Passe2142** [comment]
It sorta triggers when you wanna use an ability as xeno a pop up first then chains the ability. Would be cool if I could bind it to something else or toggle it on/off, pop up pain is real.

**Pull request merged**